app/src/main/java/ch/epfl/sdp/appart/StepCounterActivity.java

```diff
@@ -0,0 +1,295 @@
+            Log.d("PERMISSION", "Activity recognition permission granted");
+        }, () -> {
+            Log.d("PERMISSION", "Activity recognition permission refused");
+            finish();
+        });
+
+        sensorManager = (SensorManager) getSystemService(SENSOR_SERVICE);
+    }
+
+    @Override
+    public void onBackPressed() {
+        finish();
+    }
+
+    @Override
+    protected void onStart() {
+        super.onStart();
+
+        getSensorsAndAddListeners();
+
+        setTextViewStepCounter();
+        setTextViewStepDetector();
+
+        /* this resumes the activity when the user previously
+         * moved the execution in background by quitting */
+        if (startWasPressed) {
+            onStartButton(this.findViewById(R.id.startStepCount_StepCounter_Button));
+        }
+    }
+
+    /**
+     * This method starts the count.
+     * Called by the start button.
+     */
+    public void onStartButton(View view) {
+        /* needed for sensors */
+        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+
+        sensorManager = (SensorManager) getSystemService(SENSOR_SERVICE);
+
+        getSensorsAndAddListeners();
+        startButton.setVisibility(View.GONE);
+        progressBar.setVisibility(View.VISIBLE);
+        stopButton.setVisibility(View.VISIBLE);
+
+        startWasPressed = true;
+
+        /* this resets the text detector to 0 once the start button is re-pressed after stop */
+        if (stopWasPressed) {
+            textViewStepDetector.setText("0");
+            stopWasPressed = false;
+        }
+
+        setTextViewStepCounter();
+        setTextViewStepDetector();
+
+    }
+
+    /**
+     * This method stops the step count and resets all counters and states.
+     * Called by the stop button.
+     */
+    public void onStopButton(View view) {
+        startButton.setVisibility(View.VISIBLE);
+        progressBar.setVisibility(View.GONE);
+        stopButton.setVisibility(View.GONE);
+
+        stopWasPressed = true;
+
+        /* unregister listeners for step sensor signals */
+        unregisterListeners();
+
+        resetDefaultActivityValues();
+
+        /* clear the screen flag */
+        getWindow().clearFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
+    }
+
+    private void getSensorsAndAddListeners() {
+
+        if (sensorManager.getDefaultSensor(Sensor.TYPE_STEP_DETECTOR) != null) {
+            mStepDetector = sensorManager.getDefaultSensor(Sensor.TYPE_STEP_DETECTOR);
+            sensorManager.registerListener(this, mStepDetector, SensorManager.SENSOR_DELAY_NORMAL);
+            stepDetectorSensorIsAvailable = true;
+        } else {
+            /* if only the STEP_DETECTOR sensor is missing this activity will still work
+             * but with less accuracy */
+            makeText(this, "Attention: this device has no step detector sensor. This causes higher latency and less accurate step count!", Toast.LENGTH_LONG).show();
```

Comment:
Maybe you can move this in a string resource

Reply:
done 👌